### ISSUE TITLE
I would like to delete all current exercises from a language track.

### ISSUE THREAD

**Issue by piara108** [exercism/exercism.io#2720]
I was wondering if there's a way to delete all of the exercises in a language track.  For example, coming back to JavaScript and deleting all of the current iterations and start over.  It would be a nice feature when coming back to the language!

**Comment by kytrinyx** [comment]
I'd be open to having this implemented in some way.

In the meanwhile, do you want me to clear out your javascript solutions?

**Comment by piara108** [comment]
Hey Katrina, thanks for getting back to me.  Could you clear my Javascript and Ruby solutions, please?  Thanks!

Sincerely,
Piara Sandhu

**Comment by tejasbubane** [comment]
@kytrinyx I was thinking we can have this in the exercises table on profile page where a person can mark one or more exercises - something like gmail - then delete or archive. What do you think?

**Comment by kytrinyx** [comment]
Yeah, that sounds good. I'd like to move them to a deleted_iterations table rather than just deleting them outright.

**Comment by GustavoCaso** [comment]
@kytrinyx is there someone working on this one ??

**Comment by kytrinyx** [comment]
Not that I know of.

**Comment by sheekap** [comment]
I'd like this feature as well. If it's not too much trouble, would it be possible to get all my ruby exercises deleted so I can start the track afresh?

**Comment by GustavoCaso** [comment]
I'm going to try working on a solution this weekend. I like the idea @tejasbubane, an interface like gmail sounds great.

**Comment by kytrinyx** [comment]
Excellent, thanks @GustavoCaso!

**Comment by GustavoCaso** [comment]
@kytrinyx Sorry for the delay, to much work lately.

I have some questions, for the interface, would you like something like this?
![image](https://cloud.githubusercontent.com/assets/4672858/13737872/fc138e3e-e9c0-11e5-883e-36e33a9b1e5c.png)

**Comment by kytrinyx** [comment]
Yeah, that would work for me.

**Comment by GustavoCaso** [comment]
@kytrinyx would you like to move the completely `UserExercise` row to the new table `DeletedIterations` ?

What I did is add a `deleted` column to `UserExercise` and add a record to the `DeletedIterations` table, the for the table  schema is `user_id` and `exercise_id` and `timestamps`

**Comment by GustavoCaso** [comment]
Now the I think it better I guess is better to move the entire row to `DeletedIterations` so that way the user can still add new `UserExercise`, and the progress and all the information from other pages will be updated immediately

**Comment by GustavoCaso** [comment]
But what frightens me, is the data duplication

**Comment by kytrinyx** [comment]
@GustavoCaso Don't be frightened by the data duplication--the whole point is to duplicate it so that we don't lose data. The UserExercise doesn't belong in the deleted iterations, since it's not an iteration. I don't know if there's any information there that we need though. Does it matter that someone skipped an exercise? We can reconstruct the other data from the iterations.

I will probably be redoing the data model on this to use event sourcing so we never overwrite or lose any of this data.

**Comment by GustavoCaso** [comment]
@kytrinyx so are we talking about moving the necessary information from `UserExercise` and `Iterations` to the `DeletedIterations` and remove the `UserExercise` from the table ?

**Comment by kytrinyx** [comment]
If there is necessary information, then it needs to go into a separate table. I'll look at it now.

**Comment by kytrinyx** [comment]
```
                                         Table "public.user_exercises"
      Column       |            Type             |
-------------------+-----------------------------
 id                | integer                     | 
 user_id           | integer                     | 
 language          | character varying(255)      | 
 slug              | character varying(255)      | 
 iteration_count   | integer                     | 
 state             | character varying(255)      | 
 created_at        | timestamp without time zone | not null
 updated_at        | timestamp without time zone | not null
 key               | character varying(255)      | 
 archived          | boolean                     | default false
 last_iteration_at | timestamp without time zone | 
 last_activity_at  | timestamp without time zone | 
 last_activity     | character varying           | 
 fetched_at        | timestamp without time zone | 
 skipped_at        | timestamp without time zone | 
```

None of this information is crucial. I would suggest that we simply delete the row. A new one will be created as soon as they fetch or skip the exercise.

**Comment by GustavoCaso** [comment]
@kytrinyx What about the 'submissions' related with that 'UserExercise' they should be move to `DeletedIterations` ?

**Comment by kytrinyx** [comment]
Yeah, submissions are the old name for what I think about as iterations now. These are what should be moved to deleted iterations.

**Comment by kytrinyx** [comment]
This has now been implemented -- see #2801
